### DISCUSSION TITLE
variable scales & spacing for faceting

### DISCUSSION THREAD

**Discussion by nick-youngblut** [mwaskom/seaborn#3381] [Q&A]
An extremely useful feature of the ggplot2 R package is the ability to select variable scales and spacing for each facet: https://ggplot2-book.org/facet.html#sec-controlling-scales

It would be great to have the same functionality in seaborn so that one does not need to port their data from python to R just to use this feature of ggplot2.

**Comment by mwaskom** [comment]
Can you please be specific about what you are asking for, simply linking to the R documentation is not very helpful or appreciated since seaborn does not aim to provide 1:1 parity with the ggplot api. Maybe you’re looking for plotnine?

**Comment by nick-youngblut** [comment]
>since seaborn does not aim to provide 1:1 parity with the ggplot api

Sorry for not being specific enough. This is a feature request for varying x and/or y scales for faceted plots. The `space="free(_[xy])*"` feature of `facet_grid()` in ggplot2 is especially useful for faceting a bar graph (or similar) in which the number of x-axis categories differs substantially between facets. From the linked ggplot docs:

```
mpg2$model <- [reorder](https://rdrr.io/r/stats/reorder.factor.html)(mpg2$model, mpg2$cty)
mpg2$manufacturer <- [reorder](https://rdrr.io/r/stats/reorder.factor.html)(mpg2$manufacturer, -mpg2$cty)

ggplot(mpg2, aes(cty, model)) + 
  geom_point() + 
  facet_grid(manufacturer ~ ., scales = "free", space = "free") +
  theme(strip.text.y = element_text(angle = 0))
```

> Maybe you’re looking for plotnine

Seaborn is a great package, but I find myself constantly porting data over to R just to utilize some features of ggplot2. I've run into issues and limitations with plotnine (versus using ggplot2 in R), so I've somewhat avoided it.

**Comment by mwaskom** [comment]
Maybe you're looking for `sharex=False`, sorry, I really can't help too much with "here is some example ggplot code"...
